pkg/api/docs/definitions/datasources.go

```diff
@@ -200,13 +200,17 @@ import (
 //
 // Query metrics.
 //
+// Please refer to [updated API](#/ds/queryMetricsWithExpressions) instead
```

Comment:
I think that we might need to update the `queryMetricsWithExpressions` definition to make it output proper response example in open api spec because the data frames conversion to JSON is custom, but I might be wrong?

Reply:
Do you mean when it returns encoded arrow? AFAIK everything is JSON nowadays https://github.com/grafana/grafana/blob/main/pkg/api/metrics.go#L98

Reply:
Was referring to JSON, but haven't checked swagger or the generated open api spec so not sure. Mainly referring to this https://github.com/grafana/grafana-plugin-sdk-go/blob/a6e537820b7b4554637de588e87ca2d77b8cad90/backend/data.go#L74-L82 that writes the JSON in a custom way by iterating the responses/frames.

Reply:
Checked using swagger now, http://localhost:3000/swagger-ui. This might be a bigger problem than just this endpoint and potentially something we might attack later. But here it goes.

The request body example includes the following:
```json
{
  "debug": true,
  "from": "now-1h",
  "queries": [
    {
      "datasourceId": 86,
      "format": "table",
      "intervalMs": 86400000,
      "maxDataPoints": 1092,
      "rawSql": "SELECT 1 as valueOne, 2 as valueTwo",
      "refId": "A"
    }
  ],
  "to": "now"
}
```
 

The 200 response includes the following:
<details><summary>Details</summary>

```json
{
  "Responses": {
    "additionalProp1": {
      "message": "string",
      "results": {
        "additionalProp1": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp2": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp3": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        }
      }
    },
    "additionalProp2": {
      "message": "string",
      "results": {
        "additionalProp1": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp2": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp3": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        }
      }
    },
    "additionalProp3": {
      "message": "string",
      "results": {
        "additionalProp1": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp2": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        },
        "additionalProp3": {
          "dataframes": {},
          "error": "string",
          "meta": {},
          "refId": "string",
          "series": [
            {
              "name": "string",
              "points": [
                [
                  {
                    "Float64": 0,
                    "Valid": true
                  }
                ]
              ],
              "tags": {
                "additionalProp1": "string",
                "additionalProp2": "string",
                "additionalProp3": "string"
              }
            }
          ],
          "tables": [
            {
              "columns": [
                {
                  "text": "string"
                }
              ],
              "rows": [
                [
                  {}
                ]
              ]
            }
          ]
        }
      }
    }
  }
}
```
</details>